core/src/main/scala/fs2/util/syntax.scala

```diff
@@ -0,0 +1,26 @@
+package fs2.util
+
+/** Provides infix syntax for the typeclasses in the util package. */
```

Comment:
I think we should keep it public so that folks writing combinators that operate on arbitrary effect types can pick up the syntax too. Give them one less reason to write combinators fixed to `Task`.